torch/distributed/_tensor/api.py

```diff
@@ -213,6 +213,26 @@ def __repr__(self):
         # TODO: consider all_gather the local tensors for better debugging
         return f"DTensor(local_tensor={self._local_tensor}, device_mesh={self._spec.mesh}, placements={self._spec.placements})"
 
+    def __tensor_flatten__(self):
+        """
+        protocol to inform how to flatten a DTensor to local tensor
+        for PT2 tracing
+        """
+        return self._local_tensor, self._spec
+
+    @staticmethod
+    def __tensor_unflatten__(local_tensor, spec):
+        assert spec is not None
```

Comment:
nit: If this assertion is raised, should we point the user to `self._spec` being `None` in the `__tensor_flatten__()` return value?